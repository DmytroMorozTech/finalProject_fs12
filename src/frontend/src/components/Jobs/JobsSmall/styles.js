import { makeStyles } from '@material-ui/core/styles'

export default makeStyles((theme) => ({

  container: {
    display: 'flex',
    justifyContent: 'space-between',
    border: theme.border.simple,
    borderRadius: theme.shape.medium,
    backgroundColor: theme.palette.common.white,
    padding: theme.spacing(2),
    marginBottom: theme.spacing(2)
  },

  item: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    fontSize: theme.typography.h6.fontSize,
    fontWeight: theme.typography.fontWeightBold,
    color: theme.palette.grey[700],
    cursor: 'pointer'
  },

  icon: {
    fontSize: theme.typography.icons.medium.fontSize,
    display: 'flex'
  },

  title: {
    marginLeft: theme.spacing(2),
    color: theme.palette.grey[600]
  },

  button: {
    color: theme.palette.grey[600],
    '&:hover': {
      transitionDuration: '0.5s',
      backgroundColor: 'rgba(0,0,0,0.1)'
    }
  }
}))
