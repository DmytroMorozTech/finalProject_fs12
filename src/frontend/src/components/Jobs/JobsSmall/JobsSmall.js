import styles from './styles'
import React from 'react'
import BookmarkIcon from '@material-ui/icons/Bookmark'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown'
import SimpleMenu from '../../../shared/PopupMenu/PopupMenu'
import JobsAdditions from './JobsAdditions/JobsAdditions'
import SharedLinkSquare from '../../../shared/SharedLinkSquare/SharedLinkSquare'
import { Link } from 'react-router-dom'

function JobsSmall () {
  const classes = styles()

  return (
    <div className={classes.container}>
      <Link to='/jobs/my_jobs' className={classes.item} onClick={() => console.log('My Jobs')}>
        <div className={classes.icon}>
          <BookmarkIcon fontSize="inherit"/>
        </div>
        <div className={classes.title}>
          My Jobs
        </div>
      </Link>
      <SimpleMenu menuItem={
        <SharedLinkSquare title='More' icon={<ArrowDropDownIcon/>}/>
      } userData={<JobsAdditions/>}/>
    </div>
  )
}

export default JobsSmall
