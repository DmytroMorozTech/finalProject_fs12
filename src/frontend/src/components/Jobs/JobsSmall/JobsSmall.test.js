import { shallow } from 'enzyme'
import React from 'react'
import JobsSmall from './JobsSmall'

jest.mock('./styles', () => () => ({}))
jest.mock('react-router-dom', () => ({
  Link: 'a'
}))

const setUp = () => shallow(<JobsSmall/>)

describe('should render JobsSmall component', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })
  it('to match snapshot', () => {
    expect(component).toMatchSnapshot()
  })
})
