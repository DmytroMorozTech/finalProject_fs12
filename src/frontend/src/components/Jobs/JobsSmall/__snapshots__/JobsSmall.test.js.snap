// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render JobsSmall component to match snapshot 1`] = `
<div>
  <a
    to="/jobs/my_jobs"
  >
    <div>
      <Memo(ForwardRef(BookmarkIcon))
        fontSize="inherit"
      />
    </div>
    <div>
      My Jobs
    </div>
  </a>
  <SimpleMenu
    menuItem={
      <SharedLinkSquare
        icon={<Memo />}
        title="More"
      />
    }
    userData={<JobsAdditions />}
  />
</div>
`;
